src/Sparrow/Platform/Posix/SysUtils.cs

```diff
@@ -0,0 +1,160 @@
+                    Syscall.close(fd);
+                    return ulong.MaxValue;
+                }
+
+                Syscall.close(fd);
+
+                string str = null;
+                try
+                {
+                    str = Encoding.ASCII.GetString((byte*)pBuf.ToPointer(), (int)cgroupRead);
+                    cgroup = Convert.ToUInt64(str);
+                }
+                catch (Exception ex)
+                {
+
+                    if (_logger.IsInfoEnabled)
+                        _logger.Info($"couldn't convert string '{str}' to long. Will not respect container's memory limit if running under one.", ex);
+                    cgroup = ulong.MaxValue;
+                }
+            }
+            finally
+            {
+                Marshal.FreeHGlobal(pBuf);
+            }
+            return cgroup;
+        }
+
+        public static unsafe List<int> ReadAndParseRangesFromFile(string filename)
+        {
+            var fd = Syscall.open(filename, OpenFlags.O_RDONLY, FilePermissions.S_IRUSR);
+            if (fd < 0)
+            {
+                if (_logger.IsInfoEnabled)
+                    _logger.Info($"Cannot open '{filename}'. Will not respect container's number of cores if running under one.");
+                return null;
+            }
+
+            List<int> coresList;
+            UIntPtr readSize = (UIntPtr)1024;
+            IntPtr pBuf = Marshal.AllocHGlobal((int)readSize);
+            try
+            {
+                Memory.Set((byte*)pBuf, 0, 1024);
+                var cgroupRead = Syscall.read(fd, pBuf.ToPointer(), (ulong)readSize);
+                if (cgroupRead > 1000 || cgroupRead == 0) // check we are not garbadged
+                {
+                    if (_logger.IsInfoEnabled)
+                        _logger.Info(
+                            $"Got invalid number of characters ({cgroupRead}) from filename. Will not respect container's number of cores if running under one.");
+                    Syscall.close(fd);
+                    return null;
+                }
+
+                Syscall.close(fd);
+
+                string str = null;
+                try
+                {
+                    str = Encoding.ASCII.GetString((byte*)pBuf.ToPointer(), (int)cgroupRead);
+                    coresList = ParseRangesFromString(str);
+                }
+                catch (Exception ex)
+                {
+
+                    if (_logger.IsInfoEnabled)
+                        _logger.Info($"couldn't convert string '{str}' to long. Will not respect container's number of cores if running under one.", ex);
+                    return null;
+                }
+            }
+            finally
+            {
+                Marshal.FreeHGlobal(pBuf);
+            }
+            return coresList;
+        }
+
+        private static List<int> ParseRangesFromString(string str)
```

Comment:
Tons of allocations here, we are better of using the higher level API and not dealing with all this low level code.
This method shouldn't be called a lot. 

Note that I don't like it because it presume a certain file format, but you don't have any reference to what is the format or what you are reading.